scripts/seastar-json2code.py

```diff
@@ -256,68 +257,93 @@ def resolve_model_order(data):
             models.add(model_name)
     return res
 
+
 def create_enum_wrapper(model_name, name, values):
-    enum_name = model_name + "_" + name
-    res = Template("""  enum class $enum_name { $enumerators, NUM_ITEMS };
-""").substitute(enum_name=enum_name, enumerators=", ".join(values))
+    enum_name = f"{model_name}_{name}"
     wrapper = name + "_wrapper"
-    res = res + Template("""  struct $wrapper : public json::jsonable  {
-        $wrapper() = default;
-        virtual std::string to_json() const {
-            switch(v) {
-        """).substitute({'wrapper' : wrapper})
-    for enum_entry in values:
-        res = res + "      case " + enum_name + "::" + enum_entry + ": return \"\\\"" + enum_entry + "\\\"\";\n"
-    res = res + Template("""      default: return \"\\\"Unknown\\\"\";
+
+    case_clauses = "\n".join(
+        Template('''case $enum_name::$enum_entry: return "\\"$enum_entry\\"";''').substitute(
+            enum_name=enum_name, enum_entry=enum_entry) for enum_entry in values)
+    res = Template("""
+    $wrapper() = default;
+    virtual std::string to_json() const {
+        switch(v) {
+$case_clauses
```

Comment:
See how this breaks logical indentation?

I use

```python
   blah = textwrap.dedent('''\
         foo bar
              blah blah
         foo bar
   ''')

To avoid dedenting below the regular Python indentation. Also use \ to skip the first line feed.

Reply:
@avikivity thanks for your inputs. the logical indention in the python code is preserved in the latest revision. the hint on skipping the first newline is helpful, but turns out what i really need is to avoid indenting the first line, so i promote the `not_first()` predicate factory to the top-most scope, so it can be reused.